chore: adding benchmark plan

viveksingh101994 opened the pull request:
### Generate mock data
From `packages/benchmarking`
- `.env`
```
LLM_BASE_URL=
LLM_API_KEY=
LLM_CHAT_MODEL=
```
- run `pnpm generate:small`. This will then generate new dataset at `packages/benchmarking/output`

----
### Start clone MCP server
From `packages/clone-mcp-server`
- prepare the dataset by running the benchmarking script or download this https://ebee-mock-data.s3.us-west-2.amazonaws.com/output.zip and extract into `packages/benchmarking/`
- run `pnpm dev` will then start the server in `localhost:4000`
- connect to the mcp servers with below configs (may need to use ngrok incase the client can't connect to localhost
```json
{
  "name": "notion",
  "type": "streamable-http",
  "url": "http://localhost:4000/mcp/notion",
}

{
  "name": "github",
  "type": "streamable-http",
  "url": "http://localhost:4000/mcp/github",
}

{
  "name": "fathom",
  "type": "streamable-http",
  "url": "http://localhost:4000/mcp/fathom",
}

{
  "name": "slack",
  "type": "streamable-http",
  "url": "http://localhost:4000/mcp/slack",
}
```

----

### With ebee
```sh
curl --request POST \
  --url http://localhost:3000/api/mcp-servers \
  --header 'Content-Type: application/json' \
  --data '
{
  "name": "notion",
  "type": "streamable-http",
  "url": "http://localhost:4000/mcp/notion",
}
'
```
*Note:* we need to change the slack adapter to adapt this mcp server https://github.com/korotovsky/slack-mcp-server?tab=readme-ov-file so we can run ebee sync against the mock MCP server. Make sure pagination (cursor, limit) works with this mcp

viveksingh101994 committed:
chore: adding benchmark plan

buildgreatthings commented on docs/benchmark/MOCK_DATA_GENERATION_PLAN.md:
```diff
@@ -0,0 +1,659 @@
+
+**Company Profile**:
+
+- **Name**: GRagger
```
LOL

buildgreatthings commented on docs/benchmark/MOCK_DATA_GENERATION_PLAN.md:
```diff
@@ -0,0 +1,659 @@
+  // BUILD SEPARATE CONTEXTS BY CATEGORY
+
+  // Work context: ONLY work-related items
+  const workContext = {
```
this is a great start. 

I would consider nudging different types of conversations as well. For example, customer updates, recruiting, marketing, HR updates, all totally random things. The model may know to do it, also may not be a bad idea to put int he prompt

buildgreatthings commented on docs/benchmark/MOCK_DATA_GENERATION_PLAN.md:
````diff
@@ -0,0 +1,659 @@
+TIMELINE_DAYS=30
+```
+
+- **Slack**: 3,000 messages
````
<img width="1496" height="1642" alt="CleanShot 2025-12-10 at 16 27 42@2x" src="https://github.com/user-attachments/assets/a9de9b95-78d1-47f7-b318-e8ee1e27cc32" />

buildgreatthings replied:
I would asumme the PRs, notion (pages may = tasks/tickets/ product specs) may all be much more.

viveksingh101994 replied:
yes, its configurable in the code and we can increase it

buildgreatthings commented on docs/benchmark/MOCK_DATA_GENERATION_PLAN.md:
```diff
@@ -0,0 +1,659 @@
+# Mock Data Generation Plan for Multi-Source Data Integration
```
meta comments:

- what's the output format of all the files?
- how will we design the "fake" MCP server that mimics the actual MCP servers for each tool/platform to pull from our dataset?

viveksingh101994 replied:
Yes, for this im comparing the adapters we have in the server code and making it sure to have same object in output formats for each mcp call
Example: team_members in github, should return same data format in mock data and then mcp mock server will directly plug-in the generated data

CC @TranHung92

viveksingh101994 committed:
chore: adding foundation and connection logic

viveksingh101994 committed:
chore: update benchmarking mock data logic

viveksingh101994 committed:
adds readme

viveksingh101994 committed:
chore: adds unique id for meeting in fathom

viveksingh101994 committed:
chore: updating the script

viveksingh101994 committed:
chore: link step

viveksingh101994 committed:
chore: adds missing github tools

TranHung92 committed:
wip

buildgreatthings commented on packages/mock-mcp-server/README.md:
```diff
@@ -0,0 +1,361 @@
+## Available Tools
+
+### 1. `list_channels`
+
```
Can we confirm that we have all the tools for every single provider?

buildgreatthings commented on packages/mock-mcp-server/README.md:
````diff
@@ -0,0 +1,361 @@
+MOCK_DATA_PATH=../benchmarking/output/combined/data.json
+```
+
+### Claude Desktop Configuration
````
Don't think this needs to be in the.env file.

buildgreatthings commented on packages/mock-mcp-server/src/index.ts:
```diff
@@ -0,0 +1,1633 @@
+      },
+    },
+    {
+      name: "get_channel_messages",
```
I think we should treat these each as different platforms so that each one is its own MCP server when we connect from the underlying tools.

buildgreatthings commented on packages/mock-mcp-server/src/index.ts:
```diff
@@ -0,0 +1,1633 @@
+      });
+
+      return {
+        content: [
```
Also, this file is just way too long. We need to split it up into smaller ones.

TranHung92 committed:
wip

TranHung92 committed:
Merge branch 'feature/clone-mcp' into benchmark_generate_data

TranHung92 committed:
wip

TranHung92 committed:
.

viveksingh101994 committed:
chore: updating the slack messages

viveksingh101994 committed:
chore: performance

viveksingh101994 committed:
docs: add critical issues documentation for benchmark system

Add comprehensive documentation identifying and addressing critical issues
in the incremental data generation system design. This document serves as a
reference for implementation decisions and system architecture.

Key sections covered:
- Critical issues (cross-generation references, ID collisions, staleness)
- Important considerations (entity evolution, semantic extraction scaling)
- Solutions and implementation strategies
- Configuration settings and metadata management
- Status tracking and monitoring utilities

This documentation ensures all potential breaking issues are identified
upfront and provides clear guidance for maintaining system integrity across
data generations.

buildgreatthings commented on .vscode/launch.json:
```diff
@@ -0,0 +1,73 @@
+{
```
why is this needed?

buildgreatthings commented on docs/benchmark/CRITICAL_ISSUES_ADDRESSED.md:
```diff
@@ -0,0 +1,204 @@
+- Keeps "notable items" (last 50 significant issues/PRs) with IDs even when old
+- Enables cross-generation references
+
+**Implementation**: Section 3 of plan
```
where is seciton 3?

buildgreatthings commented on docs/benchmark/CRITICAL_ISSUES_ADDRESSED.md:
```diff
@@ -0,0 +1,204 @@
+---
+
+### 4. Semantic Extraction at Scale
+**Problem**: Extracting topics from 68 days of data (6,800+ messages) is slow and expensive (token limits, cost).
```
why are we extracting?

buildgreatthings commented on docs/benchmark/INCREMENTAL_GENERATION_PLAN.md:
```diff
@@ -0,0 +1,1778 @@
+
+### How It Works Now
+1. User sets `TIMELINE_DAYS=N` (e.g., 30 days)
+2. Script generates timeline from "today - N days" to "today"
```
what is we already generated last 30 days, we want the 30-60 days prior to expand the dataset length to a total of 60 days.

buildgreatthings commented on docs/benchmark/INCREMENTAL_GENERATION_PLAN.md:
````diff
@@ -0,0 +1,1778 @@
+
+**Structure**:
+```json
+{
````
this file is going to be so large, with many subpats, is a LLM going to be able to manage this well?

TranHung92 committed:
feat: slack mcp (#51)

TranHung92 committed:
feat: generate dataset (#50)

TranHung92 committed:
Merge branch 'develop' into benchmark_generate_data
